### DISCUSSION TITLE
How to reuse Dnn.React.Common components?

### DISCUSSION THREAD

**Discussion by roman-yagodin** [dnnsoftware/Dnn.Platform#4508] [Q&A] [answered]
I wonder how can one reuse React components from Dnn.React.Common (especially FolderPicker, PagePicker and FileUpload) in the "classic" WebForms, MVC or SPA modules?

**Comment by valadas** [comment, marked as answer]
We do not recommend that 3rd-party developers use that package, it is very hard to maintain it with proper semantic versioning and can have breaking changes very often. In fact it was marked obsolete in NPM.

**Reply by roman-yagodin** [reply]
I see. Can you point to any viable alternative?

**Reply by valadas** [reply]
Well the persona bar does not enforce any specific framework, it all depends what you are confortable using but it can be React, Angular, Vue, Stencil, knockout, or even plain vanilla html/css/javascript

Some of the modules only use Knockout and plain javascript if you are looking for examples.

This may change with DNN 10 as I plan to add some more reusable WebComponents that don't enforce a framework and can be reused more freely without bringing with them all kinds of other dependencies.
